src/python/pants/util/tarutil.py

```diff
@@ -7,10 +7,7 @@
 import tarfile
 from builtins import str
 
-from future.utils import implements_iterator
 
-
-@implements_iterator
 class TarFile(tarfile.TarFile):
 
   def __next__(self):
```

Comment:
This rename in a81f5cd5408497642f691925940360c6e251942d is the real problem and it's misleading, I'd suggest just reverting the rename instead of aliasing here. The `tarfile.TarFile` code does not implement `__next__` in python2 or python3, instead it implents `__iter__` with custom returned iterators that are not `tarfile.TarFile` objects.

Reply:
True. 

Even though that goes against the general Python 2-3 compatibility advice to have `next()` for Py2 and `__next__()` for Py3, you raise a good point we should mirror CPython and not implement `__next__()` since they don't.

Thanks for the review!